#!/usr/bin/env python
PACKAGE = "jpp"

import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("DR", double_t, 0, "Descriptor radius", 5.0, 0, 100)
gen.add("DQ", int_t, 0, "Radius quantization", 2, 0, 100)
gen.add("DT", int_t, 0, "Angular quantization", 1, 0, 100)
gen.add("DH", int_t, 0, "Histogram quantization", 1, 0, 100)
gen.add("LENGTH", int_t, 0, "Robot length (mm)", 400, 0, 5000)
gen.add("WIDTH", int_t, 0, "Robot width (mm)", 400, 0, 5000)
gen.add("HEIGHT", int_t, 0, "Robot height (mm)", 200, 0, 1000)
gen.add("GRID_SIZE", int_t, 0, "Grid size (mm)", 50, 10, 1000)
gen.add("CONF_POS_THRESH", double_t, 0, "Confident positive threshold", 0.75, 0, 
500)
gen.add("CONF_NEG_THRESH", double_t, 0, "Confident negative threshold", 1.2, 0, 
500)
gen.add("SAD_WINDOW_SIZE", int_t, 0, "SAD window size", 5, 1, 10)
gen.add("SPATIAL_FILTER_WINDOW", int_t, 0, "Spatial filter window size (mm)", 
100, 10, 5000)
gen.add("SPATIAL_FILTER_INC", int_t, 0, "Spatial filter grid size (mm)", 100, 
20, 5000)
gen.add("SPATIAL_FILTER_RATIO", double_t, 0, "Spatial filter ratio (s)", 0.95, 
0, 1)
gen.add("CONF_NEG_INC", int_t, 0, "Column inc size", 20, 10, 500)
gen.add("CONF_NEG_FILTER_RATIO", double_t, 0, "Conf neg filter ratio", 0.75, 0, 
1)

gen.add("Z_INC", int_t, 0, "z increment", 20, 0, 100)
gen.add("SEARCH_RANGE", int_t, 0, "Search range", 400, 0, 1000)
gen.add("ZERR_WEIGHT", double_t, 0, "z error weight", 2.0, 0, 5)
gen.add("MEDIAN_FILTER_RANGE", int_t, 0, "median filter range", 1, 0, 10)
gen.add("ROUGHNESS_THRESH", double_t, 0, "Roughness threshold", 1.8, 0, 5)

gen.add("START_X", int_t, 0, "Start x-coordinate", 500, 0, 10000)
gen.add("MAX_X", int_t, 0, "Max x-coordinate", 2000, 0, 10000)
gen.add("MAX_Y", int_t, 0, "Max y-coordinate", 300, 0, 10000)
gen.add("CONVEX_WORLD", bool_t, 0, "Convex world assumption", True)

exit(gen.generate(PACKAGE, "jpp", "Params"))
